### ISSUE TITLE
Set connection from navigator for SQL console doesn't work for tables

### ISSUE THREAD

**Issue by Matvey16** [dbeaver/dbeaver#12681]
#### System information: 
- Operating system (distribution) and version Win10
- DBeaver version 21.1.0

#### Connection specification:
- Database name and version MariaDB
- SSH

#### Describe the problem you're observing:
`Window > Set connection from navigator` doesn't work if a table or anything on any lower level is selected

#### Steps to reproduce, if exist:
1. Highlight any table in schema/container in DB Navigator that is not active in SQL Editor.
2. Click `Window > Set connection from navigator`

**Expected:** Schema for selected table is made active
**Actual:** Nothing happens.

![2021-06-01_11h18_20](https://user-images.githubusercontent.com/82543000/120290607-2d5ab100-c2cb-11eb-90b5-e632e67adabd.gif)

**Comment by flambchap** [comment]
The bug is not specific to any kind of database.

**Comment by flambchap** [comment]
Fixed.

**Comment by Matvey16** [comment]
Fix works only for currently active connection. Using `ctrl+shift+.` on an object in another connection will first set the default schema as active, and only after second `ctrl+shift+.` works as intended.

**Comment by flambchap** [comment]
Fixed once again.

**Comment by flambchap** [comment]
The fix doesn't work properly in MSSQL.

**Comment by flambchap** [comment]
It's a problem in `SQLServerExecutionContext`. When we set a new schema via `setDefaultSchema()`, the context does not switch to the current database. I suggest calling it a day and fix this problem on a separate ticket.

**Comment by Matvey16** [comment]
Verified for other databases.

New enhancement for SQL Server and Sybase #12998
